Source/VCGeneration/SplitAndVerifyWorker.cs

```diff
@@ -151,6 +153,44 @@ private async Task ProcessResult(Split split, CancellationToken cancellationToke
       callback.OnProgress?.Invoke("VCprove", splitNumber < 0 ? 0 : splitNumber, total, provenCost / (remainingCost + provenCost));
 
       if (!proverFailed) {
+        if (callback is VerificationResultCollector collector) {
```

Comment:
All the code in here will be simpler, I think, if you use the signature I suggested above.

The logic you have for identifying which assertions in a batch failed, based on the counterexamples, will still need to happen somewhere, though.

You may need to merge or rebase to incorporate the current `master` before the `VCResult` will be conveniently available.